mysys/my_init.c

```diff
@@ -170,7 +170,17 @@ my_bool my_init(void)
 
   my_progname_short= "unknown";
   if (my_progname)
+  {
+    char link_name[FN_REFLEN];
     my_progname_short= my_progname + dirname_length(my_progname);
+    /*
+      If its a link a different program that doesn't begin with mariadb
+      like mariadb-repair might link to mariadb-check.
+    */
+    if (my_readlink(link_name, my_progname, MYF(0)) == 0
+        && strncmp(my_progname_short, "mariadb", 7))
```

Comment:
swap the test, please. First `strncmp`, and only if the filename doesn't start with "mariadb" — then you try to read the symlink.